src/Adapters/Adapter.php

```diff
@@ -299,6 +299,11 @@ public function getHtml()
     public function getUrl()
     {
         $default = (string) $this->getResponse()->getUrl();
+
+        if ($this->getConfig('follow_canonical') === false) {
+            return $default;
+        }
+
```

Comment:
The option cannot be evaluated here. This function just return the canonical url, doesn't follow it. It must be checked here: https://github.com/oscarotero/Embed/blob/master/src/Embed.php#L73